webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -613,69 +632,99 @@ private static void addFmcSolutionSheet(PdfWriter docWriter, Document doc, Scram
+        float eventHeight = fitText(new Font(bf), tempMessage, new Rectangle(right-competitorInfoLeft, 100), 25f, false);
+        cb.setFontAndSize(bf, eventHeight);
         int MAGIC_NUMBER = 40; // kill me now
-        cb.showTextAligned(PdfContentByte.ALIGN_CENTER, "Fewest Moves", left+(competitorInfoLeft-left)/2, top-MAGIC_NUMBER, 0);
+        cb.showTextAligned(PdfContentByte.ALIGN_CENTER, tempMessage, left+(competitorInfoLeft-left)/2, top-MAGIC_NUMBER, 0);
         cb.endText();
+        
+        ArrayList<String> rulesList = new ArrayList<String>();
+        rulesList.add("- "+messages.getString("rule1"));
+        rulesList.add("- "+messages.getString("rule2"));
+        rulesList.add("- "+messages.getString("rule3"));
+        rulesList.add("- "+messages.getString("rule4"));
+        rulesList.add("- "+messages.getString("rule5"));
+        rulesList.add("- "+messages.getString("rule6"));
+        rulesList.add("- "+messages.getString("rule7"));
+        rulesList.add("- "+messages.getString("rule8"));
+        rulesList.add("- "+messages.getString("rule9"));
+        int maxMoves = WCA_MAX_MOVES_FMC;
+        rulesList.add("- "+String.format(messages.getString("rule10"), maxMoves));
+        rulesList.add("- "+messages.getString("rule11"));
 
-        com.itextpdf.text.List rules = new com.itextpdf.text.List(com.itextpdf.text.List.UNORDERED);
-        rules.add("Notate your solution by writing one move per bar.");
-        rules.add("To delete moves, clearly erase/blacken them.");
-        rules.add("Face moves F, B, R, L, U, and D are clockwise.");
-        rules.add("Rotations x, y, and z follow R, U, and F.");
-        rules.add("' inverts a move; 2 doubles a move. (e.g.: U', U2)");
-        rules.add("w makes a face move into two layers. (e.g.: Uw)");
-        rules.add("A [lowercase] move is a cube rotation. (e.g.: [u])");
-
-        ct.addElement(rules);
         int rulesTop = competitorInfoBottom + (withScramble ? 55 : 143);
-        ct.setSimpleColumn(left+padding, scrambleBorderTop, competitorInfoLeft-padding, rulesTop, 0, Element.ALIGN_LEFT);
-        ct.go();
-
-        rules = new com.itextpdf.text.List(com.itextpdf.text.List.UNORDERED);
-        rules.add("You have 1 hour to find a solution.");
-        rules.add("Your solution length will be counted in OBTM.");
-        int maxMoves = WCA_MAX_MOVES_FMC;
-        rules.add("Your solution must be at most " + maxMoves + " moves, including rotations.");
-        rules.add("Your solution must not be directly derived from any part of the scrambling algorithm.");
-        ct.addElement(rules);
-        MAGIC_NUMBER = 150; // kill me now
-        ct.setSimpleColumn(left+padding, scrambleBorderTop, rulesRight-padding, rulesTop-MAGIC_NUMBER, 0, Element.ALIGN_LEFT);
+        Rectangle rulesRectangle = new Rectangle(left+padding, scrambleBorderTop, competitorInfoLeft-padding, rulesTop);
+        
+        float potentialFontSize = 15f;
+        // resize rules vertically
```

Comment:
Can this logic go into fitText instead?

Reply:
That was my first try. I came into some trouble with line breaks and the part of the function that assumes [we are fitting a scramble](https://github.com/campos20/tnoodle/blob/master/webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java#L828). I decided to emulate this function using some ideas from there.

Reply:
I'll fix this.